libs/s25main/mapGenerator/HeadQuarters.cpp

```diff
@@ -90,7 +90,10 @@ namespace rttr { namespace mapGenerator {
             retries--;
         }
 
-        return success;
+        if(!success)
+        {
+            throw std::runtime_error("could not find any valid HQ position");
```

Comment:
I don't know if there're any other edge cases for random map generation I'm not aware of, but maybe it is worth to create customs exceptions for this? Something like `class NoHqPositionFound : MapCreationError` and `MapCreationError : std::exception`? Wasn't sure about it, just took the `runtime_error` which was presumably matched already to handle IO errors when writing the file?

Reply:
IMO it's fully fine. The caller may simply catch any runtime_error or even exception and report the "what", it doesn't matter much what type of error it was
However we could think about the error message. I guess it doesn't make sense to show it to the user, but maybe optionally log it (a potential range_error or so wouldn't be useful to the user, then we'd need a custom class). I'd make the messages regular sentences though, so starting with a capital letter.